qiskit/qobj/pulse_qobj.py

```diff
@@ -0,0 +1,610 @@
+            data['pulse_library'] = pulse_lib_obj
+        return cls(**data)
+
+    def __getattr__(self, name):
+        try:
+            return self._data[name]
+        except KeyError:
+            raise AttributeError('Attribute %s is not defined' % name)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseQobjConfig):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobjExperiment:
+    """A Pulse Qobj Experiment.
+
+    Each instance of this class is used to represent an individual Pulse
+    experiment as part of a larger Pulse Qobj.
+    """
+
+    def __init__(self, config=None, header=None, instructions=None):
+        """Instantiate a PulseQobjExperiment.
+
+        Args:
+            config (PulseQobjExperimentConfig): A config object for the experiment
+            header (PulseQobjExperimentHeader): A header object for the experiment
+            instructions (list): A list of :class:`PulseQobjInstruction` objects
+        """
+        if config is not None:
+            self.config = config
+        if header is not None:
+            self.header = header
+        self.instructions = instructions or []
+
+    def to_dict(self):
+        """Return a dictionary format representation of the Experiment.
+
+        Returns:
+            dict: The dictionary form of the PulseQobjExperiment.
+        """
+        out_dict = {
+            'instructions': [x.to_dict() for x in self.instructions]
+        }
+        if hasattr(self, 'config'):
+            out_dict['config'] = self.config.to_dict()
+        if hasattr(self, 'header'):
+            out_dict['header'] = self.header.to_dict()
+        return out_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a new PulseQobjExperiment object from a dictionary.
+
+        Args:
+            data (dict): A dictionary for the experiment config
+
+        Returns:
+            PulseQobjExperiment: The object from the input dictionary.
+        """
+        config = None
+        if 'config' in data:
+            config = PulseQobjExperimentConfig.from_dict(data.pop('config'))
+        header = None
+        if 'header' in data:
+            header = QasmQobjExperimentHeader.from_dict(data.pop('header'))
+        instructions = None
+        if 'instructions' in data:
+            instructions = [
+                PulseQobjInstruction.from_dict(
+                    inst) for inst in data.pop('instructions')]
+        return cls(config, header, instructions)
+
+    def __eq__(self, other):
+        if isinstance(other, PulseQobjExperiment):
+            if self.to_dict() == other.to_dict():
+                return True
+        return False
+
+
+class PulseQobjExperimentConfig:
+    """A config for a single Pulse experiment in the qobj."""
+
+    _data = {}
+
+    def __init__(self, qubit_lo_freq=None, meas_lo_freq=None):
```

Comment:
In general, arbitrary information (`**kwargs`) can be placed within here in the current version. I understand this is not necessarily a desirable feature as it can make serialization tricky.

Reply:
Oops, I missed that, nice catch. I was using the `_data = {}` pattern which was only used for the classes which had arbitrary fields, but missed the kwargs on init. Fixed in: https://github.com/Qiskit/qiskit-terra/pull/3383/commits/d476b1297243ee102a5d265470e77469932b379b